controllers/kustomization_impersonation.go

```diff
@@ -177,15 +177,15 @@ func (ki *KustomizeImpersonation) getKubeConfig(ctx context.Context) ([]byte, er
 	}
 
 	var kubeConfig []byte
-	for k := range secret.Data {
-		if k == "value" || k == "value.yaml" {
-			kubeConfig = secret.Data[k]
-			break
-		}
-	}
-
-	if len(kubeConfig) == 0 {
-		return nil, fmt.Errorf("KubeConfig secret '%s' doesn't contain a 'value' key ", secretName.String())
+	if key := ki.kustomization.Spec.KubeConfig.SecretRef.Key; key != "" {
+		kubeConfig = secret.Data[key]
```

Comment:
In retrospect also better for helm-controller.

If the user has declared their own configuration, it might be better to error out to let them know their configuration is not correct.